index.js

```diff
@@ -0,0 +1,159 @@
+let dataObject;//Declare object that contains a single key. Key's value is an array with words.
+let stringJSON;//Declare StringJSON so we could use it is submitWord();
+const inputBar = document.getElementById("addWordInput");//Add Words input bar Object.
+const clearWordsBar = document.getElementById("clearWordInput");
+const clearPassword = "yuvy";//I can use a better way, like getting this password from another database so the password isn't with the client.
+const url = "https://api.myjson.com/bins/vy0qi";//ID of the database in a variable for readability. 
+let currentButton;
+let currentButtonDOMID;
+window.onload = async function(){//Once the window loads, fetch the words from the database. Make async function beacause retrieveWords is a promise
+    let dbObj = await retrieveWords();//Get the words from the database, and whether submit mode is enabled or not
+    dataObject = dbObj;//Make the database we just fetched into local database
+    
+}
+
+inputBar.addEventListener("keyup", (event) => {//Check for when 'Enter' key is pressed and submit word to database
+    if (event.keyCode === 13) {
+     event.preventDefault();
+     submitWord(inputBar.value);//Submit the word to database
+     inputBar.value = "";//Clear the input bar
+    }
+});
+
+function retrieveWords(){//Call this function to fetch database words
+    return new Promise(async(resolve) => {//Return a promise when they are fetched
```

Comment:
No need to return a promise since you're doing await later anyways, it's redundant. Just return the json

Reply:
Technically you could/should have the button disabled until the promise is resolved so your UI doesn't freeze. 

Just use [then()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then)

Reply:
Wouldn't I have to return a promise because in submitWord() I am having to update my current local database and that could take time so I just do it asynchronously and have retrieveWords() return a promise. Otherwise it doesn't update and the UI shouldn't freeze when using await because using await still allows the UI to be on the main thread.